src/main/java/com/paweldampc/service/CarsService.java

```diff
@@ -23,31 +25,31 @@ public CarsService(CarsRepository carsRepository, CarMapper carMapper) {
     }
 
     public List<Car> findAllToRepair() {
-        List<Car> allCarsToRepair = carsRepository.getCars().stream().filter(car -> car.getRepaired().equals(false)).collect(Collectors.toList());
+        List<Car> allCarsToRepair = carsRepository.getCars().stream()
+                .filter(car -> car.getRepaired().equals(false))
+                .sorted(Comparator.comparing(Car::getCreatedDate))
+                .collect(Collectors.toList());
         return allCarsToRepair;
     }
 
     public List<Car> findAllRepaired() {
-        List<Car> allCarsRepaired = carsRepository.getCars().stream().filter(car -> car.getRepaired().equals(true)).collect(Collectors.toList());
+        List<Car> allCarsRepaired = carsRepository.getCars().stream()
+                .filter(car -> car.getRepaired().equals(true))
+                .sorted(Comparator.comparing(Car::getCreatedDate))
+                .collect(Collectors.toList());
         return allCarsRepaired;
     }
 
     public void AddCar(CarDto carDto) {
-        carsRepository.addCar(carDto.getPlateNumber(), carDto.getName(), carDto.getColor(), carDto.getYearOfProduce(), carDto.getCreated(), carDto.getRepaired());
+        carsRepository.addCar(carDto.getPlateNumber(), carDto.getName(), carDto.getColor(), carDto.getYearOfProduce(), carDto.getCreatedDate(), carDto.getRepaired());
     }
 
-
-    /*    public
-        public CarDto find (String plate){
-            Car car = carsRepository.;
-            return carMapper.toDTO(car);*/
-//    }
-    public void findByPlate(String plate) {
-        carsRepository.getCars().stream().filter(car -> car.getPlateNumber().equals(plate.toUpperCase())).forEach(car -> car.setRepaired(true));
+    public List<Car> findByPlate(String plate) {
+        carsRepository.getCars().stream()
+                .filter(car -> car.getPlateNumber().equals(plate.toUpperCase()))
+                .forEach(car -> {
+                    car.setRepaired(true);
+                    car.setCreatedDate(LocalDate.now());});
+        return null;
```

Comment:
Metoda jest zadeklarowana, że ma zwracać `List<Car>`, a zawsze zwraca null 😉 
Trzeba pilnować, żeby metoda robiła to, co mówi jej nazwa. W tym przypadku nie tylko wyszukuje auto po parametrze 'plate', ale również ustawia status 'repaired'. A więc złamana jest pierwsza zasada SOLID (Single Responsibility Principle), tyle że na poziomie metody 😉